Sources/DatadogSDKTesting/Utils/CodeOwners.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  CodeOwner.swift
+//  DatadogSDKTesting
+//
+//  Created by Ignacio Bonafonte Arruga on 15/6/21.
+//
+
+import Foundation
+
+struct CodeOwnerEntry {
+    var path: String
+    var codeowners: String
+}
+
+struct CodeOwners {
+    var ownerEntries = [CodeOwnerEntry]()
+
+    init?(workspacePath: URL) {
+        // Search on the possible locations CODEOWNER can exist
+        var location = workspacePath.appendingPathComponent("CODEOWNERS")
+        if !FileManager.default.fileExists(atPath: location.path) {
+            location = workspacePath.appendingPathComponent(".github").appendingPathComponent("CODEOWNERS")
+            if !FileManager.default.fileExists(atPath: location.path) {
+                location = workspacePath.appendingPathComponent(".docs").appendingPathComponent("CODEOWNERS")
+                if !FileManager.default.fileExists(atPath: location.path) {
+                    return nil
+                }
+            }
+        }
+
+        guard let codeOwnersContent = try? String(contentsOf: location) else {
+            return nil
+        }
+
+        self.init(content: codeOwnersContent)
+    }
+
+    init(content: String) {
+        // Parse all lines that include information
+        content.components(separatedBy: .newlines).forEach {
+            let line = $0.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+            if line.isEmpty || line.hasPrefix("#") {
+                return
+            }
+            let lineComponents = line.components(separatedBy: .whitespaces).filter { !$0.isEmpty }
+
+            let owners = lineComponents.dropFirst().joined(separator: " ")
+            if let path = lineComponents.first,
+               !owners.isEmpty
+            {
+                ownerEntries.append(CodeOwnerEntry(path: path, codeowners: owners))
+            }
+        }
+    }
+
+    func ownersForPath(_ path: String) -> String? {
+        for owner in ownerEntries.reversed() {
+            if codeOwnersWildcard(path, pattern: owner.path) {
+                return owner.codeowners
+            }
+        }
+        return nil
+    }
+
+    private func codeOwnersWildcard(_ string: String, pattern: String) -> Bool {
+        var finalPattern: String
+
+        if pattern.hasPrefix("*") || pattern.hasPrefix("/") {
+            finalPattern = pattern
+        } else {
+            finalPattern = "*" + pattern
+        }
+
+        if pattern.hasSuffix("/") {
+            finalPattern += "*"
+        }
+        let matches = genericWildcard(string, pattern: finalPattern)
```

Comment:
Doesn't it simply transform to only 3 following predicates?
* _starts with `x`_;
* _contains `x`_;
* _ends with `x`_.

If yes, maybe we don't need this expensive `NSPredicate` (I believe it uses 💸  regular expression inside).

Also, I think we don't need to build the predicate on each query, but we can pre-compile it and build it only once when reading the `CODEOWNERS` file:
```swift
struct CodeOwnerEntry {
    let codeowners: String
    let ownsPath: (String) -> Bool
}

// Parse the path expression when creating `CodeOwnerEntry` 
// - only once on loading the `CODEOWNERS` file, e.g.:
[
  CodeOwnerEntry(codeowners: "docs@example.com") { path in path.starts(with: "docs/") },
  CodeOwnerEntry(codeowners: "@doctocat") { path in path.contains(string: "/build/logs/") },
  CodeOwnerEntry(codeowners: "@js-owner") { path in path.ends(with: ".js") },
  CodeOwnerEntry(codeowners: "@global-owner1 @global-owner2") { _ in true } // this must be last
]
```

Reply:
💡 This might be easy to apply if `path` passed to `ownersForPath` is normalized (e.g. it's a relative path from repository root).

Reply:
Yes, you are true, I started using NSPredicate to simplify the wildcard handling but at the end the logic is simpler than expected, so it is a bit overkilling. I will try to rework. Thanks a lot

Reply:
Now using the String methods, it should be fast enough even for being called every time.